dymos/examples/cannonball/doc/test_doc_two_phase_cannonball.py

```diff
@@ -12,100 +12,217 @@ class TestTwoPhaseCannonballForDocs(unittest.TestCase):
+            in the ODE but instead its outputs are connected to the trajectory via
+            input design parameters.
+            """
+            def setup(self):
+                self.add_input(name='radius', val=1.0, desc='cannonball radius', units='m')
+                self.add_input(name='dens', val=7870., desc='cannonball density', units='kg/m**3')
+
+                self.add_output(name='mass', shape=(1,), desc='cannonball mass', units='kg')
+                self.add_output(name='S', shape=(1,), desc='aerodynamic reference area', units='m**2')
+
+                self.declare_partials(of='mass', wrt='dens')
+                self.declare_partials(of='mass', wrt='radius')
+
+                self.declare_partials(of='S', wrt='radius')
+
+            def compute(self, inputs, outputs):
+                radius = inputs['radius']
+                dens = inputs['dens']
+
+                outputs['mass'] = (4/3.) * dens * np.pi * radius ** 3
+                outputs['S'] = np.pi * radius ** 2
+
+            def compute_partials(self, inputs, partials):
+                radius = inputs['radius']
+                dens = inputs['dens']
+
+                partials['mass', 'dens'] = (4/3.) * np.pi * radius ** 3
+                partials['mass', 'radius'] = 4. * dens * np.pi * radius ** 2
+
+                partials['S', 'radius'] = 2 * np.pi * radius
+
+        #############################################
+        # Build the ODE class
+        #############################################
+        class CannonballODE(om.ExplicitComponent):
+            """
+            Cannonball ODE assuming flat earth and accounting for air resistance
+            """
+
+            def initialize(self):
+                self.options.declare('num_nodes', types=int)
+
+            def setup(self):
+                nn = self.options['num_nodes']
+
+                # static parameters
+                self.add_input('m', units='kg')
+                self.add_input('S', units='m**2')
+                # 0.5 good assumption for a sphere
+                self.add_input('CD', 0.5)
+
+                # time varying inputs
+                self.add_input('h', units='m', shape=nn)
+                self.add_input('v', units='m/s', shape=nn)
+                self.add_input('gam', units='rad', shape=nn)
+
+                # state rates
+                self.add_output('v_dot', shape=nn, units='m/s**2')
+                self.add_output('gam_dot', shape=nn, units='rad/s')
+                self.add_output('h_dot', shape=nn, units='m/s')
+                self.add_output('r_dot', shape=nn, units='m/s')
+                self.add_output('ke', shape=nn, units='J')
+
+                # Ask OpenMDAO to compute the partial derivatives using complex-step
+                # with a partial coloring algorithm for improved performance
+                self.declare_partials('*', '*', method='cs')
+                self.declare_coloring(wrt='*', method='cs')
```

Comment:
Don't need the declare_partials if we have declare_coloring

Reply:
Good point, since we define the ODE in that same example we should probably just tag the state rates.